plugins/obs-ffmpeg/obs-ffmpeg-mux.c

```diff
@@ -443,13 +443,19 @@ static void signal_failure(struct ffmpeg_muxer *stream)
 
 	ret = deactivate(stream, 0);
 
+#define SRT_HOST_ERROR "Failed to resolve hostname"
+
 	switch (ret) {
 	case FFM_UNSUPPORTED:
 		code = OBS_OUTPUT_UNSUPPORTED;
 		break;
 	default:
 		if (stream->is_network) {
-			code = OBS_OUTPUT_DISCONNECTED;
+			char *result = strstr(error, SRT_HOST_ERROR);
```

Comment:
I'm also not sure this is a great solution to the problem, for example if someone's internet goes down briefly then their DNS server will likely be unreachable, causing a "Failed to resolve hostname" error even though it is a situation that could be recovered from by trying again later.

Reply:
damn; you're our network guru so i'll defer to you.